test/cli/cli.test.js

```diff
@@ -1,46 +1,73 @@
+    expect(result.stdout).toContain('worker 1 started')
+    expect(result.stdout).toContain('worker 2 started')
+    expect(result.stdout).toContain('worker 1 exited')
+    expect(result.stdout).toContain('worker 2 exited')
+    expect(result.stdout).toContain('HTML Files generated in')
+  })
 
-      const nuxtGenerate = spawn(binNyc, [
-        '--reporter', 'none',
-        binGenerate, '-b', rootDir, '-w', '2', '-v'
-      ], { env })
+  test('bin/nuxt-generate with unhandled rendering error', async () => {
+    await expect(runGenerate('unhandled-error')).rejects.toSatisfy((result) => {
+      expect(result.code).toEqual(1)
+      expect(result.stderr).toContain('There were 1 unhandled page rendering errors.')
+      return true
+    })
+  })
 
-      nuxtGenerate.stdout.on('data', (data) => { stdout += data.toString() })
-      nuxtGenerate.stderr.on('data', (data) => { stderr += data.toString() })
-      nuxtGenerate.on('error', (err) => { error = err })
-      nuxtGenerate.on('close', () => {
-        resolve({ stdout, stderr, error })
-      })
+  test('bin/nuxt-generate with killed worker', async () => {
+    await expect(runGenerate('kill-error')).rejects.toSatisfy((result) => {
+      expect(result.code).toEqual(1)
+      expect(result.stderr).toContain('1 workers failed')
+      return true
     })
+  })
 
-    expect(error).toBeUndefined()
-    expect(stderr).toContain('==== Error report ====')
-    expect(stdout).toContain('Nuxt files generated')
-    expect(stdout).toContain('worker 1 started')
-    expect(stdout).toContain('worker 2 started')
-    expect(stdout).not.toContain('worker 3 started')
-    expect(stdout).toContain('generated: ')
-    expect(stdout).toContain(`${path.sep}users${path.sep}1${path.sep}index.html`)
-    expect(stdout).toContain('worker 1 exited')
-    expect(stdout).toContain('worker 2 exited')
-    expect(stdout).toContain('HTML Files generated in')
+  test('bin/nuxt-generate with a complex site', async () => {
+    await expect(runGenerate('basic')).rejects.toSatisfy((result) => {
+      expect(result.code).toEqual(1)
+      expect(result.stdout).toContain('Nuxt files generated')
+      expect(result.stdout).toContain('worker 1 started')
+      expect(result.stdout).toContain('worker 2 started')
+      expect(result.stdout).not.toContain('worker 3 started')
+      expect(result.stdout).toContain('generated: ')
+      expect(result.stdout).toContain(`${path.sep}users${path.sep}1${path.sep}index.html`)
+      expect(result.stdout).toContain('worker 1 exited')
+      expect(result.stdout).toContain('worker 2 exited')
+      expect(result.stdout).toContain('HTML Files generated in')
+      expect(result.stderr).toContain('==== Error report ====')
+      expect(result.stderr).toContain('There were 3 unhandled page rendering errors.')
+      return true
+    })
   })
 })
+
+/**
+ * @param {String} fixtureName
+ * @returns {Promise<{stdout: string, stderr: string}>}
+ */
+function runGenerate(fixtureName) {
```

Comment:
Please put helper functions in the `test/utils/` dir

Reply:
done